fabric-tag-extensions-v0/src/main/java/net/fabricmc/fabric/mixin/tag/extension/MixinMinecraftServer.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.tag.extension;
+
+import java.util.Collection;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.resource.ServerResourceManager;
+import net.minecraft.server.MinecraftServer;
+
+import net.fabricmc.fabric.impl.tag.extension.TagFactoryImpl;
+
+@Mixin(MinecraftServer.class)
+public class MixinMinecraftServer {
```

Comment:
This mixin is useless, [in my testing calling the lifecycle event `SERVER_STARTING` is sufficient](https://github.com/LambdAurora/LambdaFoxes/blob/1.17/src/main/java/dev/lambdaurora/lambdafoxes/LambdaFoxes.java#L67).

Reply:
I guess the `runServer` one can be removed, but we still need the datapack reload one though since we wanted to reload the tags before syncing packet and `END_DATAPACK_RELOAD` is called after the sync.

Reply:
> I guess the `runServer` one can be removed, but we still need the datapack reload one though since we wanted to reload the tags before syncing packet and `END_DATAPACK_RELOAD` is called after the sync.

In my testing the other one is not needed at all actually, `/reload` doesn't have the same issues as server starting.

The dynamic registry seems to not be rebuilt after tags when a reload happen, or it might be a side-effect of worldgen not being reloadable? Would need more testing.

Reply:
Looking through LambdaFoxes' code, it doesn't seem to address `Couldn't load tag {} as it is missing following references:` error that thrown when it tries to load dynamic registry tags. I [deliberately](https://github.com/goodasintended/fabric/blob/cba6f29aaefff3765ee9cc8d342c1d3e793f9128/fabric-tag-extensions-v0/src/main/java/net/fabricmc/fabric/mixin/tag/extension/MixinTagManagerLoader.java#L38) disabled loading those tags from `ServerResourceManager#reload` to make it doesn't error so `loadDynamicRegistryTags` needs to be called on both setup and reload. It's a "cosmetic" only error though, I can remove those if you guys have a problem with that part.

Reply:
I think the server would fail only if you make your tag via `RequiredTagList#add`, it'll throw `Missing required tags: ` alongside that error.

Reply:
I would move the runServer one to the lifecycle event, and let the other one as a mixin then.